inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -85,6 +88,21 @@ protected function get_home_url() {
 		return self::$home_url;
 	}
 
+	/**
+	 * Get Home protocol relative URL
+	 *
+	 * It will be used for validating URLs to keep both protocols (http or https) in sitemaps.
+	 *
+	 * @return string
+	 */
+	protected function get_home_proto_rel_url() {
```

Comment:
I'd like to suggest you use the `WPSEO_Link_Type_Classifier`. It was created for this exact purpose.

This does require you create an instance of this in the `__construct()` of this function.

As a result, you'll be able to delete this method and the static field you're setting in this method.

Reply:
I agree that's better approach. We can't create instance in constructor because it's too early. There are comments - `get_home_url`:
>  `*` Get Home URL
>  `*`
>  `*` This has been moved from the constructor because wp_rewrite is not available on plugins_loaded in multisite.
>  `*` It will now be requested on need and not on initialization.

I'm creating function `get_classifier()` similar as current function `get_image_parser()`. Also, we need to get `base_url` via `get_home_url` to keep compatibility with WPML.